[Bug Report] Incorrect typing for modelValue of DatePicker

<!-- generated by https://elementui.github.io/issue-generator DO NOT REMOVE -->

### Element Plus version
1.2.0-beta.5

### OS/Browsers version
OSX/Chrome

### Vue version
3.2.24

### Reproduction Link
https://jsfiddle.net/Lzg0e96p/4/

### Steps to reproduce
Use el-date-picker in the following manner: https://jsfiddle.net/Lzg0e96p/4/

Then try to create a build, this causes the following error:

```
 - error TS2322: Type 'string[]' is not assignable to type 'string | Date | Date[] | undefined'.
  Type 'string[]' is not assignable to type 'Date[]'.
    Type 'string' is not assignable to type 'Date'.
```

### What is Expected?
Functionally, everything works as expected, a build fails on the error above though.

So the expectation is that this command works without error:
`vue-tsc --noEmit --skipLibCheck`

### What is actually happening?
`vue-tsc --noEmit --skipLibCheck` 

causes this error:

```
 - error TS2322: Type 'string[]' is not assignable to type 'string | Date | Date[] | undefined'.
  Type 'string[]' is not assignable to type 'Date[]'.
    Type 'string' is not assignable to type 'Date'.
```

<!-- generated by https://elementui.github.io/issue-generator DO NOT REMOVE -->

--- a/packages/components/table/src/table-column/defaults.ts
+++ b/packages/components/table/src/table-column/defaults.ts
@@ -40,7 +40,7 @@ interface TableColumnCtx<T> {
     column: TableColumnCtx<T>,
     cellValue,
     index: number
-  ) => VNode
+  ) => VNode | string
   selectable: (row: T, index: number) => boolean
   reserveSelection: boolean
   filterMethod: FilterMethods<T>

--- a/packages/components/time-picker/src/common/picker.vue
+++ b/packages/components/time-picker/src/common/picker.vue
@@ -211,7 +211,7 @@ const parser = function (
   return day.isValid() ? day : undefined
 }
 
-const formatter = function (date: Date, format: string, lang: string) {
+const formatter = function (date: number | Date, format: string, lang: string) {
   return isEmpty(format) ? date : dayjs(date).locale(lang).format(format)
 }
 

--- a/packages/components/time-picker/src/common/props.ts
+++ b/packages/components/time-picker/src/common/props.ts
@@ -61,7 +61,7 @@ export const timePickerDefaultProps = {
     default: () => ({}),
   },
   modelValue: {
-    type: [Date, Array, String] as PropType<string | Date | Date[]>,
+    type: [Date, Array, String] as PropType<string | Date | (number | Date)[]>,
     default: '',
   },
   rangeSeparator: {

